pkg/webhooks/workload_webhook.go

```diff
@@ -75,6 +75,11 @@ func (w *WorkloadWebhook) Default(ctx context.Context, obj runtime.Object) error
 			wl.Spec.PodSets[i].MinCount = nil
 		}
 	}
+
+	// If a deactivated workload is re-activated, we need to reset the RequeueState.
```

Comment:
I'm not convinced this works, because the apiserver doesn't generally allow you to update both the status and spec.

Did you add an integration test for it?